usr.bin/look/look.c

```diff
@@ -132,21 +138,35 @@ main(int argc, char *argv[])
 
 	match = 1;
 
-	do {
-		if ((fd = open(file, O_RDONLY, 0)) < 0 || fstat(fd, &sb))
+#ifndef WITHOUT_CAPSICUM
+	cap_rights_init(&rights, CAP_MMAP_R, CAP_READ, CAP_FSTAT);
```

Comment:
I used WITHOUT_CAPSICUM assuming it is required to handle cases where the kernel is built without Capsicum support, rather than strictly for tools in contrib. Could you clarify if my understanding is off? I'd appreciate any insights!

Reply:
On FreeBSD if you build a kernel without the CAPSICUM support the `cap_*` functions still exists however they will fail and return ENOSYS.
To avoid checking errno you can use capsicum helpers.
The WITHOUT_CAPSICUM is for platforms that aren't aware of capsicum's existence, like Linux. So we have to add them to contrib source code which isn't managed by FreeBSD but not in the code that is dedicated for FreeBSD.

Reply:
Thanks, I didn't know that. I'll take care of it.